src/backings/structural/container.ts

```diff
@@ -152,23 +153,29 @@ export class ContainerStructuralHandler<T extends ObjectLike> extends Structural
     const fieldType = this._type.fields[fieldName];
     return fieldType.hashTreeRoot(value[fieldName]);
   }
-  fromJson(data: Json): T {
+  fromJson(data: Json, options?: IJsonOptions): T {
     if (typeof data !== "object") {
       throw new Error("Invalid JSON container: expected Object");
     }
     const value = {} as T;
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
-      if ((data as Record<string, Json>)[fieldName] === undefined) {
+
+      const expectedCase = options ? options.case : null;
+      const expectedFieldName = toExpectedCase(fieldName, expectedCase);
+      if ((data as Record<string, Json>)[expectedFieldName] === undefined) {
         throw new Error(`Invalid JSON container field: expected field ${fieldName} is undefined`);
       }
-      value[fieldName as keyof T] = fieldType.fromJson((data as Record<string, Json>)[fieldName]);
+      value[fieldName as keyof T] = fieldType.fromJson(
+        (data as Record<string, Json>)[expectedFieldName]
+      );
     });
     return value;
   }
-  toJson(value: T): Json {
+  toJson(value: T, options?: IJsonOptions): Json {
     const data = {} as Record<string, Json>;
+    const expectedCase = options ? options.case : null;
     Object.entries(this._type.fields).forEach(([fieldName, fieldType]) => {
-      data[fieldName] = fieldType.toJson(value[fieldName as keyof T]);
+      data[toExpectedCase(fieldName, expectedCase)] = fieldType.toJson(value[fieldName as keyof T]);
```

Comment:
call to toJson should have options as second arg
we should make sure that all recursively called toJson and fromJson have options passed in (here and in array/vector/list)